### PR TITLE
test(backend): freeze MCP context contract

### PR DESCRIPTION
**Pull request by akadlec**
> 👍 Codex review passed on commit `75669a0a2` with no major issues.

## Summary

- lock whole-home and scoped `get_home_context` routing and MCP envelope preservation
- freeze exact MCP payloads, numeric caps, truncation behavior, and deterministic source ordering
- characterize disabled-but-visible compatibility and reject hidden-device property history
- assert security, weather, energy, timeseries, discovery, resource, and installation contracts
- document the provider-neutral extraction contract and complete the final Phase 0 checklist item

## Why

Phase 1 will move domain queries behind a provider-neutral home-context module. These tests freeze the current MCP protocol behavior first, so Buddy can reuse domain services without silently changing MCP output, limits, scope, policy, visibility, or transport semantics.

## Validation

- final MCP/domain regression slice: 51 suites, 628 tests, 2 snapshots passed
- focused review-fix suites, ESLint, and Prettier checks passed
- backend type-check passed
- `git diff --check` passed
- all review threads resolved; Codex review found no major issues

### CODE REVIEW

File: tasks/plans/plan-buddy-adaptive-context.md [hunk 1]
```diff
@@ -1184,7 +1184,8 @@ regenerate checked-in artifacts from backend Swagger sources; never edit generat
 - [x] Record the eager small-window budget violation as a measured baseline fixture or opt-in expected-failure evaluation
       excluded from normal CI. Keep the Phase 0 verification suite green; promote the fixture to a required passing
       bounded-request assertion when Phase 7 switches the production conversation path.
-- [ ] Confirm existing MCP context outputs/limits in tests before extracting them.
+- [x] Confirm existing MCP context outputs/limits in tests before extracting them; the frozen compatibility contract is
+      recorded in [buddy-adaptive-context-mcp-baseline.md](./buddy-adaptive-context-mcp-baseline.md).
```
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/plan-buddy-adaptive-context.md:1188]: Addressed in 064a45e50. McpContextService now has exact whole-home and scoped composite payload assertions, including every section, collection order, mapped field name, and limits object. The read-adapter whole-home fixture also contains and exactly asserts spaces, devices, scenes, weather, energy, security, and limits.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 2]
```diff
@@ -530,6 +692,39 @@ describe('McpContextService', () => {
 		expect(energy.getSpaceSummary).not.toHaveBeenCalled();
 	});
 
+	it('rejects energy ranges beyond the MCP compatibility limit before querying', async () => {
+		const from = '2026-01-01T00:00:00.000Z';
+		const to = new Date(Date.parse(from) + (MCP_MAX_ENERGY_RANGE_DAYS + 1) * 24 * 60 * 60 * 1000).toISOString();
+
+		await expect(service.getEnergySummary(from, to)).rejects.toThrow(
+			`may not exceed ${MCP_MAX_ENERGY_RANGE_DAYS} days`,
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts]: Addressed in d8e0855b2. The suite now independently pins all 17 documented numeric limits to literal values. Security, forecast, and timeseries cases use literal over-limit inputs and literal caps; energy explicitly proves 31 days is accepted and 32 days is rejected.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 3]
```diff
@@ -384,6 +530,41 @@ describe('McpContextService', () => {
 		expect(result.limits).toEqual(expect.objectContaining({ scenes_truncated: true }));
 	});
 
+	it('preserves disabled visible devices and scenes in the MCP snapshot', async () => {
+		devices.findVisibleSummaryPage.mockResolvedValue({
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:735]: Fixed in 14fa0168f. I added regression coverage at the actual domain-query boundaries: `DevicesService.findVisibleSummaryPage` now proves disabled visible devices are returned while the query filters only `hidden = false`, and `ScenesService.findSummaryPage` proves disabled scenes remain in unscoped summaries with no enabled predicate. The combined domain + MCP contract slice passes (4 suites, 96 tests), with focused ESLint and `git diff --check` clean.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 4]
```diff
@@ -553,6 +796,27 @@ describe('McpContextService', () => {
...
+		timeseries.queryTimeseriesStrict.mockResolvedValue({
+			bucket: '1h',
+			points: Array.from({ length: 501 }, (_, index) => ({ time: index, value: index })),
+		});
+
+		const result = await service.getPropertyTimeseries(
+			'property-id',
+			'2026-08-01T00:00:00.000Z',
+			'2026-08-14T00:00:00.000Z',
+			'1h',
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:1127]: Fixed in 0901ce9a1. The contract suite now uses literal timestamps to prove an exact 14-day timeseries range succeeds and a 14-day-plus-one-second range fails with `may not exceed 14 days`. It also asserts the rejected call is stopped before a second property/storage query. Focused context/read tests pass (2 suites, 45 tests), with ESLint, Prettier, and `git diff --check` clean.

File: apps/backend/src/modules/devices/services/devices.service.spec.ts [hunk 5]
```diff
@@ -272,6 +272,26 @@ describe('DevicesService', () => {
 	});
 
 	describe('findVisibleSummaryPage', () => {
+		it('includes disabled devices while excluding hidden devices at the query boundary', async () => {
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/devices/services/devices.service.spec.ts:275]: Fixed in 724792092. `DevicesService.findVisibleSummaryById` is now exercised with `enabled: false, hidden: false`; the test asserts the real query uses the hidden predicate and no enabled predicate. `McpContextService.getDeviceState` also proves the disabled flag survives the direct-read mapping.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 6]
```diff
@@ -129,6 +138,46 @@ describe('McpContextService', () => {
...
 	});
 
+	it('freezes the numeric MCP context and discovery compatibility limits', () => {
+		expect({
+			spaces: MCP_MAX_CONTEXT_SPACES,
+			devices: MCP_MAX_CONTEXT_DEVICES,
+			channelsPerDevice: MCP_MAX_CHANNELS_PER_DEVICE,
+			propertiesPerChannel: MCP_MAX_PROPERTIES_PER_CHANNEL,
+			scenes: MCP_MAX_CONTEXT_SCENES,
+			writableProperties: MCP_MAX_WRITABLE_PROPERTIES,
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:152]: Fixed in 724792092. Registered `list_writable_properties` coverage now proves 100 actionable candidates return all 100 with `truncated: false`, while 101 candidates return the first 100, omit the 101st, and report `truncated: true`. The combined boundary slice passes (4 suites, 110 tests), with ESLint, Prettier, and `git diff --check` clean.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 7]
```diff
@@ -172,6 +221,103 @@ describe('McpContextService', () => {
...
+					name: 'Morning',
+					category: 'generic',
+					enabled: true,
+					triggerable: true,
+					primary_space_id: 'room-1',
+				},
+			],
+			weather: null,
+			energy: null,
+			security: null,
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts]: Fixed in 410fe0141. Both exact composite fixtures now exercise successful optional-domain composition with a fixed clock. Whole-home and scoped snapshots assert complete mapped weather, energy scope/range/summary, and security status/alerts/truncation/last-event payloads rather than null placeholders.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 8]
```diff
@@ -129,6 +138,46 @@ describe('McpContextService', () => {
...
+		}).toEqual({
+			spaces: 50,
+			devices: 100,
+			channelsPerDevice: 20,
+			propertiesPerChannel: 40,
+			scenes: 50,
+			writableProperties: 100,
+			writableCandidates: 500,
+			triggerScenes: 50,
+			triggerSpaces: 50,
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:173]: Fixed in 410fe0141. Registered `list_trigger_targets` tests now pin literal query limits of 50 for scenes and spaces, prove 50/50 returns are not truncated, and prove 50 returned from totals of 51 set the corresponding per-collection truncation flag. Combined focused tests pass (3 suites, 66 tests), with ESLint, Prettier, and `git diff --check` clean.

File: apps/backend/src/modules/devices/services/devices.service.spec.ts [hunk 9]
```diff
@@ -339,9 +359,13 @@ describe('DevicesService', () => {
...
 			jest.spyOn(repository, 'createQueryBuilder').mockReturnValue(queryBuilderMock);
 
-			await expect(service.findVisibleSummaryById(mockDevice.id)).resolves.toEqual(visibleDevice);
+			await expect(service.findVisibleSummaryById(mockDevice.id)).resolves.toEqual(
+				expect.objectContaining({ id: visibleDevice.id, enabled: false, hidden: false }),
+			);
 			expect(queryBuilderMock.leftJoinAndSelect).toHaveBeenCalledTimes(1);
 			expect(queryBuilderMock.leftJoinAndSelect).toHaveBeenCalledWith('device.deviceZones', 'deviceZones');
+			expect(queryBuilderMock.andWhere).toHaveBeenCalledWith('device.hidden = :hidden', { hidden: false });
+			expect(queryBuilderMock.andWhere).not.toHaveBeenCalledWith(expect.stringContaining('enabled'), expect.anything());
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/devices/services/devices.service.spec.ts]: Fixed in 7304391d3. The page and by-ID query tests now collect the first argument from every `where` and `andWhere` call, then assert no recorded predicate contains `enabled`. This catches both one-argument and parameterized filters.

File: tasks/plans/plan-buddy-adaptive-context.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/plan-buddy-adaptive-context.md:1188]: Fixed in 7304391d3. `getDeviceState` now has an exact complete service fixture covering identity, room/zones, status, channel/property metadata, current value/timestamps/trend, and truncation flags. `getPropertyTimeseries` now exactly asserts from/to/bucket/points/truncation and storage arguments. Registered read-tool tests prove exact argument forwarding and unchanged enveloped data for both. Combined focused tests pass (3 suites, 95 tests), with ESLint, Prettier, and `git diff --check` clean.

File: tasks/plans/buddy-adaptive-context-mcp-baseline.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+Successful tools retain the established envelope fields:
+
+```text
+installation, request_id, tool, observed_at, data
+```
+
+Tool arguments are passed through without silently changing whole-home versus scoped behavior. Policy, installation
+identity, client credentials, audit records, MCP deadlines, sanitized errors, and protocol registration remain in the MCP
+adapter. Phase 1 may replace the mapper behind `McpContextService`, but must not move these transport concerns into the
+shared query module.
```
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/buddy-adaptive-context-mcp-baseline.md:76]: Fixed in 81ff0afb8. A registered `get_energy_summary` regression now passes literal `{ from, to, space_id }`, asserts positional forwarding as `getEnergySummary(from, to, spaceId)`, and proves the exact returned object is preserved in `structuredContent.data`. Focused Jest (18 tests), ESLint, Prettier, and `git diff --check` pass.

File: tasks/plans/plan-buddy-adaptive-context.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/plan-buddy-adaptive-context.md:1188]: Fixed in 164c83675. Exact registered resource tests now freeze both surfaces with a fixed clock: `smart-panel://home/context` proves unscoped forwarding and the complete `{ contents: [{ uri, mimeType, text }] }` JSON wrapper, while `smart-panel://spaces/{spaceId}/snapshot` proves `spaceId` forwarding and the same exact compatible wrapper/payload. Focused Jest (20 tests), ESLint, Prettier, and `git diff --check` pass.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 11]
```diff
@@ -172,6 +225,250 @@ describe('McpContextService', () => {
...
+			},
+			devicesTruncated: false,
+			channelsTruncated: false,
+			propertiesTruncated: false,
+		});
+
+		await expect(service.getHomeContext('room-1')).resolves.toEqual({
+			scope: { type: 'space', id: 'room-1', name: 'Kitchen' },
+			spaces: [{ id: 'room-1', name: 'Kitchen', type: SpaceType.ROOM, parent_id: null, device_count: 0 }],
+			devices: [],
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:455]: Fixed in f080cd5b5. A scoped room snapshot fixture now returns 100 devices from total 101, asserts `findVisibleDeviceSummariesBySpace('room-1', 100)`, preserves all 100 returned devices, and reports `limits.devices_truncated: true`.

File: apps/backend/src/modules/mcp/tools/mcp-read-tool.service.spec.ts [hunk 12]
```diff
@@ -60,13 +88,20 @@ describe('McpReadToolService', () => {
...
+		resourceTemplateCallbacks = new Map();
 		resourceListCallback = undefined;
 		registerTool = jest.fn((name: string, _config: unknown, callback: ToolCallback) => {
 			callbacks.set(name, callback);
 		});
-		registerResource = jest.fn((name: string, _uri: unknown, _config: unknown, callback: ResourceCallback) => {
-			resourceCallbacks.set(name, callback);
-		});
+		registerResource = jest.fn(
+			(name: string, _uri: unknown, _config: unknown, callback: ResourceCallback | ResourceTemplateCallback) => {
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/tools/mcp-read-tool.service.spec.ts]: Fixed in f080cd5b5. The registration harness now captures URI arguments. It asserts `home-context` registers exact `smart-panel://home/context`, and `space-snapshot` registers an SDK `ResourceTemplate` whose public template string is exact `smart-panel://spaces/{spaceId}/snapshot`. Focused tests pass (2 suites, 51 tests), with ESLint, Prettier, and `git diff --check` clean.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 13]
```diff
@@ -129,6 +142,46 @@ describe('McpContextService', () => {
...
+			securityDevices: MCP_MAX_SECURITY_DEVICES,
+			securityChannelsPerDevice: MCP_MAX_SECURITY_CHANNELS_PER_DEVICE,
+			securityPropertiesPerChannel: MCP_MAX_SECURITY_PROPERTIES_PER_CHANNEL,
+			forecastDays: MCP_MAX_FORECAST_DAYS,
+			timeseriesRangeDays: MCP_MAX_TIMESERIES_RANGE_DAYS,
+			timeseriesPoints: MCP_MAX_TIMESERIES_POINTS,
+			energyRangeDays: MCP_MAX_ENERGY_RANGE_DAYS,
+		}).toEqual({
+			spaces: 50,
+			devices: 100,
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:166]: Fixed in 9bd6e2fc3. A literal whole-home boundary fixture now asserts `findVisibleSummaryPage(100)`, returns 100 devices from total 101, and reports `limits.devices_truncated: true`.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 14]
```diff
@@ -411,15 +771,41 @@ describe('McpContextService', () => {
...
+					severity: 'warning',
+					message: `Alert ${index}`,
+				})),
+			},
+			devicesTruncated: false,
+			channelsTruncated: false,
+			propertiesTruncated: false,
+		});
+
+		const result = await service.getSecurityStatus();
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:821]: Fixed in 9bd6e2fc3. The public direct `getSecurityStatus` regression now asserts its own `getBoundedStatus(100, 10, 20)` invocation. Focused context tests pass (32 tests), with ESLint, Prettier, and `git diff --check` clean.

File: apps/backend/src/modules/mcp/tools/mcp-target-discovery-tool.service.spec.ts [hunk 15]
```diff
@@ -256,6 +330,41 @@ describe('McpTargetDiscoveryToolService', () => {
...
+
+	it('returns only 100 of 101 actionable writable properties and marks the discovery result as truncated', async () => {
+		const targets = writableProperties(101);
+		channelsPropertiesService.findWritableCandidates.mockResolvedValue({
+			properties: targets,
+			total: 101,
+		});
+		service.register(server(), authInfo([McpCapability.WRITE]));
+
+		const result = await callbacks.get('list_writable_properties')?.({}, requestContext([McpCapability.WRITE]));
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/tools/mcp-target-discovery-tool.service.spec.ts:406]: Fixed in a789596b6. Writable discovery now asserts the exact ordered array of two actionable targets around filtered candidates, including property ID/name/category, device ID/name, channel ID/name/category, data type, resolved unit, format, step, and invalid values (including the null contract).

File: apps/backend/src/modules/mcp/tools/mcp-target-discovery-tool.service.spec.ts [hunk 16]
```diff
@@ -184,6 +184,80 @@ describe('McpTargetDiscoveryToolService', () => {
...
+		const result = await callbacks.get('list_trigger_targets')?.({}, requestContext([McpCapability.TRIGGER]));
+		const data = result?.structuredContent.data as {
+			scenes: Array<{ scene_id: string }>;
+			truncated: { scenes: boolean; spaces: boolean };
+		};
+
+		expect(scenesService.findTriggerableSummaryPage).toHaveBeenCalledWith(50);
+		expect(data.scenes).toHaveLength(50);
+		expect(data.scenes[49]?.scene_id).toBe(targets[49]?.id);
+		expect(data.truncated.scenes).toBe(true);
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/tools/mcp-target-discovery-tool.service.spec.ts:269]: Fixed in a789596b6. The mixed trigger fixture enables both providers, pins both literal query limits, and exactly asserts complete scene/space mapped arrays plus independent `truncated: { scenes: true, spaces: false }` for totals 51 and 50. Focused Jest (21 tests), ESLint, Prettier, and `git diff --check` pass.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 17]
```diff
@@ -411,15 +794,41 @@ describe('McpContextService', () => {
...
+
+		expect(result.active_alerts).toHaveLength(20);
+		expect(result).toEqual(expect.objectContaining({ active_alerts_count: 23 }));
+		expect(result.alerts_truncated).toBe(true);
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts]: Fixed in f6f9ff235. The 23-alert fixture now supplies complete distinct alerts and exactly asserts the ordered mapped prefix `alert-0` through `alert-19`, including type, severity, timestamp, acknowledgement, source device, and message.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 18]
```diff
@@ -530,6 +961,42 @@ describe('McpContextService', () => {
...
+		weather.getWeather.mockResolvedValue(weatherResult);
+
+		const primary = await service.getWeather();
+		const explicit = await service.getWeather('location-id');
+
+		expect(weather.getPrimaryWeather).toHaveBeenCalledTimes(1);
+		expect(weather.getWeather).toHaveBeenCalledWith('location-id');
+		expect(primary.forecast).toHaveLength(5);
+		expect(explicit).toEqual(primary);
+		expect(explicit).toEqual(expect.objectContaining({ location_id: 'location-id', location: 'Prague' }));
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:1034]: Fixed in f6f9ff235. The seven-entry forecast fixture now exactly asserts the ordered first five mapped entries (August 15–19), preventing reverse/tail selection.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 19]
```diff
@@ -538,19 +1005,81 @@ describe('McpContextService', () => {
...
+
+		const result = await service.getPropertyTimeseries(
+			'property-id',
+			'2026-08-01T00:00:00.000Z',
+			'2026-08-14T00:00:00.000Z',
+			'1h',
+		);
+
+		expect(result.points).toHaveLength(500);
+		expect(result.truncated).toBe(true);
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts]: Fixed in f6f9ff235. The 501-point fixture now exactly asserts the ordered retained point array `0..499`, pinning both the boundary values and prefix order. Focused context Jest (32 tests), ESLint, Prettier, and `git diff --check` pass.

File: apps/backend/src/modules/mcp/tools/mcp-read-tool.service.spec.ts [hunk 20]
```diff
@@ -93,6 +132,120 @@ describe('McpReadToolService', () => {
 			'get_security_status',
 		]);
 		expect(registerResource).toHaveBeenCalledTimes(3);
+		expect(registeredResourceUris.get('home-context')).toBe('smart-panel://home/context');
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/tools/mcp-read-tool.service.spec.ts:136]: Fixed in 2f1b9228a. The captured resource registration map now also asserts `installation` is exactly `smart-panel://installation`, alongside the home and scoped snapshot URI assertions.

File: apps/backend/src/modules/devices/services/devices.service.spec.ts [hunk 21]
```diff
@@ -272,6 +272,30 @@ describe('DevicesService', () => {
 	});
 
 	describe('findVisibleSummaryPage', () => {
+		it('includes disabled devices while excluding hidden devices at the query boundary', async () => {
+			const disabledVisibleDevice = toInstance(MockDevice, { ...mockDevice, enabled: false, hidden: false });
+			const queryBuilderMock: any = {
+				leftJoinAndSelect: jest.fn().mockReturnThis(),
+				where: jest.fn().mockReturnThis(),
+				andWhere: jest.fn().mockReturnThis(),
+				orderBy: jest.fn().mockReturnThis(),
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/devices/services/devices.service.spec.ts:281]: Fixed in 2f1b9228a. The real visible-device summary query boundary now asserts literal `orderBy('device.name', 'ASC')`, preserving capped device selection/output order.

File: apps/backend/src/modules/scenes/services/scenes.service.spec.ts [hunk 22]
```diff
@@ -9,6 +9,31 @@ import { SceneActionsService } from './scene-actions.service';
 import { ScenesService } from './scenes.service';
 
 describe('ScenesService', () => {
+	it('includes disabled scenes in summary queries', async () => {
+		const disabledScene = { id: 'disabled-scene', name: 'Disabled scene', enabled: false } as SceneEntity;
+		const query = {
+			select: jest.fn().mockReturnThis(),
+			orderBy: jest.fn().mockReturnThis(),
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/scenes/services/scenes.service.spec.ts:16]: Fixed in 2f1b9228a. The real scene summary query boundary now asserts literal `orderBy('scene.name', 'ASC')`, preserving capped scene selection/output order. Combined focused Jest (3 suites, 72 tests), ESLint, Prettier, and `git diff --check` pass.

File: tasks/plans/plan-buddy-adaptive-context.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/plan-buddy-adaptive-context.md:1188]: Fixed in 4981174a8. The real `ChannelsService.findSummaryPage` query-boundary spec now asserts literal `orderBy('channel.name', 'ASC')`, preserving which channels survive the cap and their external order. Focused Jest (13 tests), ESLint, Prettier, and `git diff --check` pass.

File: apps/backend/src/modules/mcp/tools/mcp-read-tool.service.spec.ts [hunk 23]
```diff
@@ -93,6 +132,121 @@ describe('McpReadToolService', () => {
 			'get_security_status',
 		]);
 		expect(registerResource).toHaveBeenCalledTimes(3);
+		expect(registeredResourceUris.get('installation')).toBe('smart-panel://installation');
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/tools/mcp-read-tool.service.spec.ts:135]: Fixed in 51cba642e. A successful registered installation-resource test now proves `getInstallation([READ], endpoint)` receives live policy capabilities and auth endpoint, and exactly asserts the bare installation JSON resource wrapper, URI, MIME type, and payload.

File: apps/backend/src/modules/mcp/tools/mcp-target-discovery-tool.service.spec.ts [hunk 24]
```diff
@@ -177,13 +205,107 @@ describe('McpTargetDiscoveryToolService', () => {
 		const data = result?.structuredContent.data as {
 			scenes: Array<Record<string, unknown>>;
 			spaces: Array<Record<string, unknown>>;
+			truncated: { scenes: boolean; spaces: boolean };
 		};
 
-		expect(data.scenes).toHaveLength(1);
-		expect(data.spaces).toEqual([expect.objectContaining({ modes: ['off', 'on', 'work', 'relax', 'night'] })]);
+		expect(scenesService.findTriggerableSummaryPage).toHaveBeenCalledWith(50);
+		expect(spacesService.findLightingTriggerSummaryPage).toHaveBeenCalledWith(50);
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/tools/mcp-target-discovery-tool.service.spec.ts:212]: Fixed in 51cba642e. Domain query tests now pin triggerable scene ordering as `scene.name ASC`, then `scene.id ASC`, and lighting-space ordering as `space.displayOrder ASC`, `space.name ASC`, then `space.id ASC`.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 25]
```diff
@@ -449,18 +873,40 @@ describe('McpContextService', () => {
...
+				last_changed: '2026-08-06T12:00:00.000Z',
+			},
+			channels: [
+				{
+					id: 'channel-id',
+					name: 'Light',
+					category: 'light',
+					properties: [
+						{
+							id: 'property-id',
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:895]: Fixed in 51cba642e. The real bounded-property query test now asserts the exact window clause `ORDER BY COALESCE(property."name", ''), property."id"`. Combined focused Jest (4 suites, 97 tests), ESLint, Prettier, and `git diff --check` pass.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 26]
```diff
@@ -530,6 +976,48 @@ describe('McpContextService', () => {
...
 
+	it('rejects energy ranges beyond the MCP compatibility limit before querying', async () => {
+		const from = '2026-01-01T00:00:00.000Z';
+		const atLimit = '2026-02-01T00:00:00.000Z';
+		const beyondLimit = '2026-02-02T00:00:00.000Z';
+		energy.getSummary.mockResolvedValue({ totalConsumptionKwh: 1 });
+
+		await expect(service.getEnergySummary(from, atLimit)).resolves.toEqual(
+			expect.objectContaining({ totalConsumptionKwh: 1 }),
+		);
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts]: Fixed in e44c576bf. The accepted explicit 31-day home range now exactly asserts scope, from/to, the complete energy summary payload, and exact `getSummary(new Date(from), new Date(to))` arguments; the >31-day rejection and single-query guarantee remain. Focused Jest (32 tests), ESLint, Prettier, and `git diff --check` pass.

File: tasks/plans/plan-buddy-adaptive-context.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/plan-buddy-adaptive-context.md:1188]: Fixed in 9d42f962b. The successful registered `get_security_status` fixture now supplies and preserves the complete mapped security object: states/severity/count/critical, full alert metadata, all truncation/state flags, and `last_event`, asserted by identity and deep equality. Focused Jest (21 tests), ESLint, Prettier, and `git diff --check` pass.

File: tasks/plans/buddy-adaptive-context-mcp-baseline.md [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+
+## Phase 1 Regression Rule
+
+Run the focused MCP service/tool specs before and after each extraction slice. Existing external output, ordering, totals,
```
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/buddy-adaptive-context-mcp-baseline.md:80]: Addressed in c352595e9. The domain spec now pins the exact writable-candidate ordering chain: device.name ASC, channel.name ASC, property.name ASC, and property.id ASC.

File: tasks/plans/plan-buddy-adaptive-context.md [hunk 28]
```diff
@@ -1184,7 +1184,8 @@ regenerate checked-in artifacts from backend Swagger sources; never edit generat
 - [x] Record the eager small-window budget violation as a measured baseline fixture or opt-in expected-failure evaluation
       excluded from normal CI. Keep the Phase 0 verification suite green; promote the fixture to a required passing
       bounded-request assertion when Phase 7 switches the production conversation path.
-- [ ] Confirm existing MCP context outputs/limits in tests before extracting them.
+- [x] Confirm existing MCP context outputs/limits in tests before extracting them; the frozen compatibility contract is
```
[the remark this answers is not in the document]
Comment (akadlec) [reply tasks/plans/plan-buddy-adaptive-context.md:1187]: Addressed in c352595e9. The bounded security-device query spec now pins the exact ORDER BY device."name", device."id" clause together with the literal scoped parameters.

File: apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts [hunk 29]
```diff
@@ -538,19 +1042,81 @@ describe('McpContextService', () => {
...
+			service.getPropertyTimeseries('property-id', '2026-08-01T00:00:00.000Z', '2026-08-15T00:00:01.000Z', '1h'),
+		).rejects.toThrow('may not exceed 14 days');
+		expect(properties.findOne).toHaveBeenCalledTimes(1);
+		expect(timeseries.queryTimeseriesStrict).toHaveBeenCalledTimes(1);
+	});
+
+	it('maps the complete direct property-timeseries contract for a visible device', async () => {
 		properties.findOne.mockResolvedValue({
 			id: 'property-id',
 			channel: { device: { hidden: false } },
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/mcp/services/mcp-context.service.spec.ts:1077]: Addressed in 22afa090c. The direct timeseries contract now rejects a property owned by a hidden device with the existing not-found message and proves queryTimeseriesStrict is never called.

File: apps/backend/src/modules/devices/services/channels.service.spec.ts [hunk 30]
```diff
@@ -259,6 +259,7 @@ describe('ChannelsService', () => {
 				total: 25,
 			});
 			expect(queryBuilderMock.innerJoin).toHaveBeenCalledWith('channel.device', 'device');
+			expect(queryBuilderMock.orderBy).toHaveBeenCalledWith('channel.name', 'ASC');
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/modules/devices/services/channels.service.spec.ts:262]: Addressed in 75669a0a2. The bounded-channel domain test now pins the exact ROW_NUMBER window ordering clause, ORDER BY channel."name", channel."id", together with the cap parameters.
